sixpack/server.py

```diff
@@ -232,12 +234,29 @@ def on_client_experiments(self, request, client_id):
 
     @service_unavailable_on_connection_error
     def on_experiment_details(self, request, name):
-        exp = Experiment.find(name, redis=self.redis)
+        api_key = request.args.get('api_key', None)
+        exp = Experiment.find(api_key, name, redis=self.redis)
         if exp is None:
             return json_error({'message': 'experiment not found'}, request, 404)
 
         return json_success(exp.objectify_by_period('day', True), request)
 
+    @service_unavailable_on_connection_error
+    def on_user_experiment_alternatives(self, request, name):
+        api_key = request.args.get('api_key', None)
+        start = request.args.get('start') or 1
+        end = request.args.get('end') or 5000
+        exp = Experiment.find(api_key, name, redis=self.redis)
+        if exp is None:
+            return json_error({'message': 'experiment not found'}, request, 404)
+        user_alternatives = experiment_user_alternatives(
+            api_key, exp, redis=self.redis, start=start, end=end)
+        resp = {
+            'user_alternatives': user_alternatives
```

Comment:
see if we can use this structure `[{'user_id': 124, 'alternative_name': 'some_alternative'},...]`